docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,149 @@
+= Hibernate for Google Cloud Spanner
+
+This is a dialect compatible with https://hibernate.org/orm/releases/5.4/[Hibernate 5.4] for the Google Cloud Spanner database service.
+The `SpannerDialect` produces SQL, DML, and DDL statements for most common entity types and relationships using standard Hibernate and Java Persistence annotations.
+
+Please see the following sections for important details and differences due to Cloud Spanner's features and data model.
+
+=== Quick Set-Up
+
+Maven coordinates for the dialect:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-spanner-hibernate-dialect</artifactId>
+  <version> TBD </version>
+</dependency>
+----
+
+Maven coordinates for the official Cloud Spanner Driver:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-spanner</artifactId>
+  <version> TBD </version>
+</dependency>
+----
+
+Configuring the `SpannerDialect` and a Cloud Spanner Driver class is typical of all Hibernate dialects in the `hibernate.properties` file:
+
+----
+hibernate.dialect=com.google.cloud.spanner.hibernate.SpannerDialect
+hibernate.connection.driver_class=com.google.cloud.spanner.jdbc.JdbcDriver
+hibernate.connection.url=jdbc:cloudspanner:/projects/{INSERT_PROJECT_ID}/instances/{INSERT_INSTANCE_ID}/databases/{INSERT_DATABASE_ID}
+----
+
+The service account JSON credentials file's location should be in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+
+The dialect and driver are compatible with all values (`create`, `create-drop`, and `update`) of the `hibernate.hbm2ddl.auto` setting.
+
+
+=== Spanner Dialect
```

Comment:
I think it might be nice to just have a specific section or table of unsupported features. Currently, you kind of have to sift through the page to find what is not supported.